Server/mods/deathmatch/logic/CConsoleCommands.cpp

```diff
@@ -1278,32 +1278,53 @@ bool CConsoleCommands::DebugScript(CConsole* pConsole, const char* szArguments,
 bool CConsoleCommands::Help(CConsole* pConsole, const char* szArguments, CClient* pClient, CClient* pEchoClient)
 {
     // Help string
-    std::string strHelpText = "Available commands:\n\n";
-
-    // Loop through all added commands
-    int                                    iCount = 0;
-    list<CConsoleCommand*>::const_iterator iter = pConsole->CommandsBegin();
-    for (; iter != pConsole->CommandsEnd(); iter++)
+    if (!szArguments)
     {
-        // Add a new line every third command
-        if (iCount == 3)
+        std::string strHelpText = "Available commands:\n\n";
+        pEchoClient->SendConsole("help [command]");
+
+        // Loop through all added commands
+        int                                    iCount = 0;
+        list<CConsoleCommand*>::const_iterator iter = pConsole->CommandsBegin();
+        for (; iter != pConsole->CommandsEnd(); iter++)
         {
```

Comment:
Can be:
```cpp
for (auto iter = pConsole->CommandsBegin(); iter != pConsole->CommandsEnd(); iter++)
```
But, my way to do it is to just add a new method to pConsole:
```cpp
const auto& CommandsList() const { return m_commands; }
```
Then just a range based for loop:
```cpp
for (const auto& command : pConsole->CommandsList())
```

Reply:
Also, just realized, this: 
```cpp
for (const auto& command : pConsole->CommandsList())
```
could be:
```cpp
for (auto command : pConsole->CommandsList())
```
A reference to a pointer is really dumb.

Reply:
```
const auto& CommandsList() { return m_commands; }
```
no need for `const`

Reply:
works fine anyway, commiting in a sec